query/interface.go

```diff
@@ -43,7 +56,7 @@ func defaultQueryOptions() *queryOptions {
 	return &queryOptions{
 		timeout:    defaultQueryTimeout,
 		encoding:   defaultQueryEncoding,
-		cancelChan: nil,
+		numRetries: defaultNumRetries,
```

Comment:
This appears to be breaking behavior for _all_ callers: previously the query functions would retry indefinitely, while now they'll give up after two attempts.